src/Microsoft.SqlTools.ServiceLayer/Management/Common/DataContainer.cs

```diff
@@ -1187,35 +1188,47 @@ internal static System.Security.SecureString BuildSecureStringFromPassword(strin
         private static XmlDocument CreateDataContainerDocument(ConnectionInfo connInfo, bool databaseExists)
         {
             string xml = string.Empty;
+            var serverNameNode = new XElement("servername");
+            serverNameNode.Value = connInfo.ConnectionDetails.ServerName.ToUpper(CultureInfo.InvariantCulture);
+
+            var connectionMonikerNode = new XElement("connectionmoniker");
+            connectionMonikerNode.Value = string.Format("{0} (SQLServer, user = {1})", serverNameNode.Value, connInfo.ConnectionDetails.UserName);
+
+            var urnNode = new XElement("urn");
+            urnNode.Value = string.Format("Server[@Name = '{0}']", serverNameNode.Value);
```

Comment:
You still need to URN escape the string here

Reply:
is this what you are expecting here?

Urn.EscapeString(serverNameNode.Value)